README.md

```diff
@@ -52,34 +52,19 @@ interface Promise
     /**
      * Registers a callback to be invoked when the promise is resolved.
      *
-     * The callback receives `null` as first parameter and `$value` as second parameter on success. It receives the
-     * failure reason as first parameter and `null` as second parameter on failure.
-     *
      * If the promise is already resolved, the callback MUST be executed immediately.
      *
-     * Warning: If you use type declarations for `$value`, be sure to make them accept `null` in case of failures.
-     *
-     * @param callable(\Throwable|\Exception|null $exception, mixed $value) $onResolved Callback to be executed.
+     * @param callable(mixed $reason, mixed $value) <`$reason` shall be `null` on success, `$value` shall be `null` on failure>
      *
      * @return void
      */
     public function when(callable $onResolved);
 }
 ```
 
-The `when()` method MUST accept at least one argument:
```

Comment:
This paragraph was removed because the docblock contains prototype and description, no need to repeat.